src/main/scala/nl/knaw/dans/easy/fedora2vault/EasyFedora2vaultApp.scala

```diff
@@ -15,10 +15,68 @@
  */
 package nl.knaw.dans.easy.fedora2vault
 
-import scala.util.{ Success, Try }
+import java.io.InputStream
 
-class EasyFedora2vaultApp(configuration: Configuration)  {
+import better.files.File
+import com.yourmediashelf.fedora.client.FedoraClient
+import nl.knaw.dans.bag.v0.DansV0Bag
+import nl.knaw.dans.easy.fedora2vault.Command.FeedBackMessage
+import nl.knaw.dans.easy.fedora2vault.FoXml.{ getEmd, _ }
+import resource.{ ManagedResource, managed }
 
+import scala.util.{ Failure, Success, Try }
+import scala.xml.{ Node, XML }
 
+class EasyFedora2vaultApp(configuration: Configuration) {
+  private lazy val fedoraClient = new FedoraClient(configuration.fedoraCredentials)
 
+  def simpleTransform(datasetId: DatasetId, outputDir: File): Try[FeedBackMessage] = {
+    for {
+      foXml <- getFoXmlInputStream(datasetId).map(XML.load).tried
+      depositor <- getOwner(foXml)
+      bag <- DansV0Bag.empty(outputDir).map(_.withEasyUserAccount(depositor))
+      emd <- getEmd(foXml)
+      _ <- bag.addMetadata(emd, "emd.xml")
+      _ <- getDdm(foXml)
+        .map(bag.addMetadata(_, "ddm.xml")).getOrElse(Success(())) // TODO EASY-2683
+      _ <- getAgreementsXml(foXml)
+        .map(bag.addMetadata(_, "agreements.xml")).getOrElse(Success(())) // TODO ?
```

Comment:
please check the locations of your files. For example, this one should be at:

```suggestion
        .map(bag.addMetadata(_, "depositor-info/agreements.xml")).getOrElse(Success(())) // TODO ?
```